ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectra.java

```diff
@@ -307,4 +314,65 @@ public void processDepositReceipts(
           false);
     }
   }
+
+  @Override
+  protected void applyDepositToValidatorIndex(
+      final MutableBeaconState state,
+      final Bytes32 withdrawalCredentials,
+      final boolean signatureAlreadyVerified,
+      final int validatorIndex,
+      final UInt64 amount) {
+    final MutableBeaconStateElectra stateElectra = MutableBeaconStateElectra.required(state);
+    final SszMutableList<PendingBalanceDeposit> pendingBalanceDeposits =
+        MutableBeaconStateElectra.required(state).getPendingBalanceDeposits().createWritableCopy();
+    pendingBalanceDeposits.append(
+        schemaDefinitionsElectra
+            .getPendingBalanceDepositSchema()
+            .create(SszUInt64.of(UInt64.fromLongBits(validatorIndex)), SszUInt64.of(amount)));
+    stateElectra.setPendingBalanceDeposits(pendingBalanceDeposits);
+    if (signatureAlreadyVerified
+        && predicatesElectra.isCompoundingWithdrawalCredential(withdrawalCredentials)) {
+      beaconStateMutatorsElectra.switchToCompoundingValidator(stateElectra, validatorIndex);
+    }
+  }
+
+  @Override
+  protected void addValidatorToRegistry(
+      final MutableBeaconState state,
+      final BLSPublicKey pubkey,
+      final Bytes32 withdrawalCredentials,
+      final UInt64 amount) {
+    final Validator validator = getValidatorFromDeposit(pubkey, withdrawalCredentials);
+    LOG.debug("Adding new validator with index {} to state", state.getValidators().size());
+    state.getValidators().append(validator);
+    final int validatorIndex = validatorsUtil.getValidatorIndex(state, pubkey).orElseThrow();
+
+    state.getBalances().appendElement(UInt64.ZERO);
```

Comment:
it might make no difference here, but shouldn't we be using the `stateElectra` to append this change to the balance?

Reply:
the balance is just on the base class, but yep can do either... maybe its easier to see what's going on if we're only mutating the electra state, i'll update.